diff --git "a/mingw-w64-libraries/winpthreads/src/thread.c" "b/mingw-w64-libraries/winpthreads/src/thread.c"
index 2eb76db..2df4e79 100644
--- "a/mingw-w64-libraries/winpthreads/src/thread.c"
+++ "b/mingw-w64-libraries/winpthreads/src/thread.c"
@@ -117,6 +117,7 @@ SetThreadName (DWORD dwThreadID, LPCSTR szThreadName)
      {
      }
 #else
+#if _WIN32_WINNT >= _WIN32_WINNT_WINXP
    /* Without a debugger we *must* have an exception handler,
     * otherwise raising an exception will crash the process.
     */
@@ -124,6 +125,7 @@ SetThreadName (DWORD dwThreadID, LPCSTR szThreadName)
    if ((!IsDebuggerPresent ()) && (SetThreadName_VEH_handle == NULL))
 #else
    if (!IsDebuggerPresent ())
+#endif
 #endif
      return;
 
@@ -717,6 +719,7 @@ int
 pthread_num_processors_np(void) 
 {
   int r = 0;
+  #if _WIN32_WINNT >= _WIN32_WINNT_WINXP
   DWORD_PTR ProcessAffinityMask, SystemAffinityMask;
 
   if (GetProcessAffinityMask(GetCurrentProcess(), &ProcessAffinityMask, &SystemAffinityMask))
@@ -724,6 +727,7 @@ pthread_num_processors_np(void)
       for(; ProcessAffinityMask != 0; ProcessAffinityMask >>= 1)
 	r += (ProcessAffinityMask & 1) != 0;
     }
+  #endif
   /* assume at least 1 */
   return r ? r : 1;
 }
@@ -733,6 +737,7 @@ pthread_num_processors_np(void)
 int
 pthread_set_num_processors_np(int n) 
 {
+#if _WIN32_WINNT >= _WIN32_WINNT_WINXP
   DWORD_PTR ProcessAffinityMask, ProcessNewAffinityMask = 0, SystemAffinityMask;
   int r = 0; 
   /* need at least 1 */
@@ -751,6 +756,9 @@ pthread_set_num_processors_np(int n)
       SetProcessAffinityMask (GetCurrentProcess (),ProcessNewAffinityMask);
     }
   return r;
+#else
+  return 1;
+#endif
 }
 
 int
