diff --git "a/mingw-w64-crt/Makefile.am" "b/mingw-w64-crt/Makefile.am"
index 89efc48..9779d44 100644
--- "a/mingw-w64-crt/Makefile.am"
+++ "b/mingw-w64-crt/Makefile.am"	
@@ -23,7 +23,7 @@ AM_CPPFLAGS=$(sysincludes)
 AM_CFLAGS=-pipe -std=gnu99 -D_CRTBLD -D_WIN32_WINNT=0x0f00 -D__MSVCRT_VERSION__=0x700 -D__USE_MINGW_ANSI_STDIO=0 @IMAGEBASE_CFLAGS@ @CFGUARD_CFLAGS@ @ADD_C_CXX_WARNING_FLAGS@ @ADD_C_ONLY_WARNING_FLAGS@
 AM_CXXFLAGS=@ADD_C_CXX_WARNING_FLAGS@ @ADD_CXX_ONLY_WARNING_FLAGS@
 CPPFLAGSARM32=-mfpu=vfpv3
-CPPFLAGS32=-m32 -masm=att
+CPPFLAGS32=-m32 -masm=att -march=pentium3
 CPPFLAGS64=-m64 -masm=att
 AM_DLLTOOLFLAGS=-k --as=$(AS) --output-lib $@
 DLLTOOLFLAGSARM32=-m arm

diff --git "a/mingw-w64-crt/Makefile.in" "b/mingw-w64-crt/Makefile.in"
index 9fcefe4..06f914c 100644
--- "a/mingw-w64-crt/Makefile.in"
+++ "b/mingw-w64-crt/Makefile.in"	
@@ -10742,7 +10742,7 @@ AM_CPPFLAGS = $(sysincludes)
 AM_CFLAGS = -pipe -std=gnu99 -D_CRTBLD -D_WIN32_WINNT=0x0f00 -D__MSVCRT_VERSION__=0x700 -D__USE_MINGW_ANSI_STDIO=0 @IMAGEBASE_CFLAGS@ @CFGUARD_CFLAGS@ @ADD_C_CXX_WARNING_FLAGS@ @ADD_C_ONLY_WARNING_FLAGS@
 AM_CXXFLAGS = @ADD_C_CXX_WARNING_FLAGS@ @ADD_CXX_ONLY_WARNING_FLAGS@
 CPPFLAGSARM32 = -mfpu=vfpv3
-CPPFLAGS32 = -m32 -masm=att
+CPPFLAGS32 = -m32 -masm=att -march=pentium3
 CPPFLAGS64 = -m64 -masm=att
 AM_DLLTOOLFLAGS = -k --as=$(AS) --output-lib $@ $(am__append_1) \
 	$(am__append_2)
